_includes/scenarios/integrate/request-and-response-introduction.md

```diff
@@ -0,0 +1,50 @@
+- [FreeTextRequestItem]({% link _docs_integrate/requests-and-requestitems.md %}#freetextrequestitem)s can be used to send a free text to a contact, that in turn can accept this with a free text as well
+- [ProposeAttributeRequestItem]({% link _docs_integrate/requests-and-requestitems.md %}#proposeattributerequestitem)s can be used to ask the peer for Attributes, already suggesting answers, e.g. asking for an address and setting the country of the organization as default
+- [ReadAttributeRequestItem]({% link _docs_integrate/requests-and-requestitems.md %}#readattributerequestitem)s can be used to ask a peer for Attributes, e.g. asking for an address
+- [ShareAttributeRequestItem]({% link _docs_integrate/requests-and-requestitems.md %}#shareattributerequestitem)s can be used to share own IdentityAttributes with a contact, e.g. sharing the own DisplayName when requesting a new Relationship
+
+In case multiple RequestItems should be answered jointly, you can combine them to a [RequestItemGroup]({% link _docs_integrate/data-model-overview.md %}#requestitemgroup).
+RequestItems or RequestItemGroups are then embedded in a Request, which can be sent to the other Identity.
+This happens either via a Message (see [Requests over Messages]({% link _docs_integrate/requests-over-messages.md %})) or via a Template (see [Requests over Templates]({% link _docs_integrate/requests-over-templates.md %})).
+
+Once the other Identity receives a Request, they can decide whether they want to accept or reject the corresponding RequestItems or RequestItemGroups.
+Consequently, the associated [ResponseItem]({% link _docs_integrate/data-model-overview.md %}#responseitem)s are created.
+They can either be a [AcceptResponseItem]({% link _docs_integrate/data-model-overview.md %}#acceptresponseitem) or a special variant of it, a [RejectResponseItem]({% link _docs_integrate/data-model-overview.md %}#rejectresponseitem) or an [ErrorResponseItem]({% link _docs_integrate/data-model-overview.md %}#errorresponseitem) in case the enmeshed Runtime detects a problem.
+If additional information is necessary to accept a Request, e.g. an Attribute is queried with a ReadAttributeRequestItem, it is attached to the AcceptResponseItem.
+In the given example, it will be a part of the returned ReadAttributeAcceptResponseItem.
+For every RequestItem or RequestItemGroup, exactly one ResponseItem or [ResponseItemGroup]({% link _docs_integrate/data-model-overview.md %}#responseitemgroup) will be created, respectively.
+The [Response]({% link _docs_integrate/data-model-overview.md %}#response), then, is wrapped in a [ResponseWrapper]({% link _docs_integrate/data-model-overview.md %}#responsewrapper), which holds additional information about the corresponding Request, so that it can be processed correctly once it is returned.
+
```

Comment:
I'm also missing the point how the parameters [(see here for an example one)](https://enmeshed.eu/integrate/requests-and-requestitems#freetextrequestitem-response-parameters) are used in the whole process.

Reply:
I guess that is because I don't fully understand them. In your example it says: Response Parameters - Item Properties - To reject this RequestItem a RejectResponseItem will be transferred. And: Response Parameters - Parameters - To reject this RequestItem you can send { "accept": false } as parameters. Why are there these two options? Also, I can't find the accept-parameter in [this overview](https://lucid.app/lucidchart/d191a08c-e1b0-4eca-9531-819f084fc197/edit?invitationId=inv_85c13f1e-4124-40da-85ea-a9f20eda6221&page=V_1ZPbLAhXaS#).

Reply:
Moreover, I looked into the code and couldn't find the "accept" parameter. Is this still current?

Reply:
Side note: we've clarified this topic last Friday. The accept parameter comes out of the Runtime library as a parameter for the usecase.

Reply:
I think it comes from consumption. See https://github.com/nmshd/cns-consumption/blob/main/src/modules/requests/itemProcessors/proposeAttribute/AcceptProposeAttributeRequestItemParameters.ts

Reply:
I guess the "accept" parameter should be discussed on the future "Respond to incoming requests" page.